### ISSUE TITLE
Cop Cuffing 

### ISSUE THREAD

**Issue by MrKennyG** [HollywoodHeists/public#113]
At the moment i believe cuffing is done by a hotkey shortcut which if done incorretly can cause major desync issues and look really weird plus... a bit powerful. Would be nice for cops to have to taze suspects then use the F1 Menu to access a cuffing action this will help rid those desync issues and make things not only look better but keep it from being too strong and give the criminal an idea of when to expect a cuff break option instead of being pulled 5 feet from where you are back to the cop.

**Comment by ChugThis** [comment]
I’d rather not taze every person I need to put in cuffs. Hard cuffing is a hot key, cuff is also available to use from our inventory. Can’t remember if there is a f1 tool for it. Tackling creates the same issue of “desync”. At what point are you being cuffed when it causes issues? Sounds like someone is cuffing you while running which is not allowed

**Comment by ChugThis** [comment]
Alternatively, can equip it in one of the 6 quickbar options which would then also need to be disabled

**Comment by ChugThis** [comment]
For the actual cuffing, can add a policy to make sure actions are called. “Going for cuffs” etc

**Comment by MrKennyG** [comment]
I mean thats literally what the tazer is for imo. Otherwise it would jump from nightstick to pistol in escalation. Tazing gives the officer time to approach the target and access an F1 menu to cuff. At the moment everyone can do the cuff animation whether you have cuffs or not and when crims do get cuffs there will not be a "rule" for them to not abuse cuffing people. This hotkey option just needs to go and the tazer / f1 menu option fixes this. Makes things more balanced and look better on both ends. Now giving chase and tazing right away is at the discretion of RP the cops wants to give.

**Comment by ChugThis** [comment]
I disagree. And I’m aware of the escalation. Cuffing someone who is fleeing will continue to be against policy and will punish those that do it in RP. Will let the guys in charge make the changes. Tazing right away is also not something that just happens. I’ve also always had a hot key for cuffs and never had an issue, so I may be biased against scrolling through a menu while trying to access your handcuffs. If you say we need to get rid of it, then can also for now ask that we don’t use the hot key and use the other options. Still don’t know why the desync is an issue between tackling and tasing, or if someone actually complies and just sits there shouldn’t be an issue either. Don’t want people getting cuffed from 5 feet away like you have said

**Comment by Bergsbrew** [comment]
Personally I think we should hold off on tazing. With how our tazer works taze spamming could easily get out of control. Personally I'd like to see tackles being more common than tazing. If we were to get away from the shortcut I don't fully see how that will fix desync. It will always be there. I've been announcing my cuffs and haven't run into an issue. If we are going to change cuffing to f1 radial it needs to be easily accessible because going through multiple menus to cuff someone just isnt going to happen when people are as squirrely as they are. LEO is getting mixed signals on tazing, some people think we did it too much so we calmed down and now youre saying we need to taze more.

**Comment by MrKennyG** [comment]
I think getting it on the main F1 wheel for cuffs is a MUST if thats how it changes. Taze spamming can be limited by making it need a reload after a certain amount of shots. I don't mind tackling but will say it can be super abused atm for crims. I could tackle a cop over and over everytime they try to stand technically.

**Comment by MrKennyG** [comment]
This is an example of what i mean here.   https://www.twitch.tv/tbnrkenworth/clip/OddLongCobblerBigBrother-aDiPagTabfPvIgjA?filter=clips&range=7d&sort=time

**Comment by DeliriousHoot** [comment]
There definitely needs to be a change to cuffing. that desync is brutal and with how fast it is you might as well not even try. I think a system where the first cuff is easier to break should be worked in with the second and 3rd cuff being considerably harder (4th cuff would be insta cuff with no chance of breaking). I do think tazing or tackling should be utilized more (both would need changes made as both systems as they stand are a bit weird) tazing would be fine if the effect didn't last quite as long

**Comment by Bergsbrew** [comment]
ATM tackling is the WAY more fair option. We have to get close and it doesnt last as long as a taze. Announcing cuffs will be part of the official SOP now. I still fail to see how changing it to the F1 will eliminate desync. I agree there needs to be changes and balances made, but maybe I need clarification on why you think it HAS to be f1.

**Comment by YouTri** [comment]
Coming from both sides Crim/LEO I think its fine the way it is imo. I don't see Tasing giving enough time to go through F1 -> Work -> Police Actions -> Cuff enough time. It sure doesn't for just tackle.

**Comment by MrKennyG** [comment]
> Coming from both sides Crim/LEO I think its fine the way it is imo. I don't see Tasing giving enough time to go through F1 -> Work -> Police Actions -> Cuff enough time. It sure doesn't for just tackle.

Ya if Cuff Action is added it needs to be on "MAIN" f1 wheel for cops. Cuffing as a hot key will always be too strong and leads to weird situations. The reason i suggested tazing is because it allows for a few seconds of letting things sync up and gives the cop time to access an F1 wheel where as tackling is too quick and makes things weird such as the clip above shows. Imagine that with someone who has high ping happening.

**Comment by Bergsbrew** [comment]
What we can do in the meantime is try and wait a second or two to let things sync up and see how that goes.

**Comment by MrKennyG** [comment]
> What we can do in the meantime is try and wait a second or two to let things sync up and see how that goes.

That works in theory but would require tazing. Tackling doesn't allow the time needed. Which brings me back to the original start of this ticket.

**Comment by Bergsbrew** [comment]
I think there are inconsistencies in tackling however. I have tackled someone and had plenty of time to wait and get a cuff. Other times not so much. This is all a balancing act right now.

So hows something like this sound? tazers should need to reload. Tackling could use a SLIGHT time buff. Breaking cuffs can use some tuning to make it slightly easier. LEOS will now always announce cuffs locally. Cuffs on immediate f1 for leos

**Comment by DeliriousHoot** [comment]
^this seems good to me. obviously the dsync issue needs to legitimately be fixed no matter what changes are made.

**Comment by MrKennyG** [comment]
> I think there are inconsistencies in tackling however. I have tackled someone and had plenty of time to wait and get a cuff. Other times not so much. This is all a balancing act right now.
> 
> So hows something like this sound? tazers should need to reload. Tackling could use a SLIGHT time buff. Breaking cuffs can use some tuning to make it slightly easier. LEOS will now always announce cuffs locally. Cuffs on immediate f1 for leos

Sounds good. I think cops should have SEARCH / CUFFF / Panick (just killed)  all on QUICK F1 Menu

**Comment by JessNikki** [comment]
> > I think there are inconsistencies in tackling however. I have tackled someone and had plenty of time to wait and get a cuff. Other times not so much. This is all a balancing act right now.
> > So hows something like this sound? tazers should need to reload. Tackling could use a SLIGHT time buff. Breaking cuffs can use some tuning to make it slightly easier. LEOS will now always announce cuffs locally. Cuffs on immediate f1 for leos
> 
> Sounds good. I think cops should have SEARCH / CUFFF / Panick (just killed) all on QUICK F1 Menu

If this is like another qbcore server I've been on there might be a /binds option you can bind CUFF/SEARCH/ESCORT to your F keys, they'll show you unassigned ones and you can choose which action to bind to those F keys, no need to have cuffs in your hot bar
